[docs] [CDCSDK] Updating CDC docs with missing limitations (#12752)
yugabyte/yugabyte-db: docs/content/preview/explore/change-data-capture/debezium-connector-yugabytedb.md

* Reads the transaction log.
* Streams change event records to Kafka topics.

{{< note title="Note" >}}

> 
> We use a custom record extractor in order to make the sinks understand the format we are sending the data in i.e. `YBExtractNewRecordState`. For example, if you are using a JDBC Sink Connector, you need to add two more properties to the sink configuration:
> 
> | Property | Value |
> | :--- | :--- |
> | `transforms` | `unwrap` |
> | `transforms.unwrap.type` | `io.debezium.connector.yugabytedb.transforms.YBExtractNewRecordState` |
> 
> More about `YBExtractNewRecordState` is mentioned in the next section.

Reviewer: ```suggestion
We use a custom record extractor (`YBExtractNewRecordState`) so that the sinks understand the format in which we are sending the data. For example, if you are using a JDBC sink connector, you need to add two more properties to the sink configuration:

| Property | Value |
| :------- | :---- |
| `transforms` | `unwrap` |
| `transforms.unwrap.type` | `io.debezium.connector.yugabytedb.transforms.YBExtractNewRecordState` |

See the following section for more details on `YBExtractNewRecordState`.
```

{{< /note >}}

#### `YBExtractNewRecordState` SMT

> Unlike the Debezium Connector for PostgreSQL, we only send the `after` image of the "set of columns" that are modified. PostgreSQL sends the complete `after` image of the row which has changed. So by default if the column was not changed, it is not a part of the payload we send and the default value is set to `null`.

Reviewer: ```suggestion
The YugabyteDB Debezium connector sends an `after` image containing _only modified columns_. (The PostgreSQL Debezium connector's `after` image contains _all_ columns of a modified row). By default, if the column hasn't changed, it's not a part of the payload, and the default value is set to `null`.
```

> Now, to differentiate between the case where a column was set `null` or was not modified, we change the value type to a struct. In this structure, an unchanged column will be `{'value': null}`, whereas a changed column that was set to `null` will be `{'value': null, 'set': true}`.

Reviewer: ```suggestion
To differentiate between the case where a column is set to `null` and the case in which it's not modified, we change the value type to a struct. In this structure, an unchanged column is `{'value': null}`, whereas the column changed to a null value is `{'value': null, 'set': true}`.
```

> We have provided an SMT (Single Message Transformer for Kafka) `YBExtractNewRecordState` that interprets these values and sends the record in the format i.e removes the unmodified columns from the JSON message. In this way, the record which is transformed by 'YBExtractNewRecordState' is compatible with all the sink implementations. We took this approach so that the schema doesn't change with each new record and it can work with "Schema Registry". For example, if we keep changing the record to only include the value of columns then the schema of each record will be different (the total number unique schemas will be a result of making all possible combinations of columns) and thus we would require sending 'schema' with all the records. This will not work with "Schema Registry" where once a schema is registered, the record can only contain payload with schema 'version' and Kafka will make sure that it is interpreted correctly.

Reviewer: I'm not sure I completely follow the logic, so please make sure my rewording here makes sense!
```suggestion
A schema registry requires that, once a schema is registered, records must contain only payloads with that schema version. If you're using a schema registry, the YugabyteDB Debezium connector's approach can be problematic, as the schema may change with every message. For example, if we keep changing the record to only include the value of modified columns, the schema of each record will be different (the total number unique schemas will be a result of making all possible combinations of columns) and thus would require sending a schema with every record.

To avoid this problem when you're using a schema registry, use the `YBExtractNewRecordState` SMT (Single Message Transformer for Kafka), which interprets these values and sends the record in the correct format (by removing the unmodified columns from the JSON message). Records transformed by `YBExtractNewRecordState` are compatible with all sink implementations. This approach ensures that the schema doesn't change with each new record and it can work with a schema registry.
```

{{< /note >}}

{{< note title="Note" >}}

> If you have a YugabyteDB cluster with SSL enabled, all you need to do is obtain the root certificate and provide the path of the file to the `database.sslrootcert` configuration property. You can follow these links to get the certificates to your universe:

Reviewer: ```suggestion
If you have a YugabyteDB cluster with SSL enabled, need to obtain the root certificate and provide the path of the file in the `database.sslrootcert` configuration property. You can follow these links to get the certificates for your universe:
```

> * [Local deployments](../../secure/tls-encryption/_index)
> * [YB Anywhere](../../yugabyte-platform/security/enable-encryption-in-transit#connect-to-a-ysql-endpoint-with-tls)
> * [YB Managed](../../yugabyte-cloud/cloud-secure-clusters/cloud-authentication#download-your-cluster-certificate)

Reviewer: Links should always have a trailing slash, even if you're following the final path element with an anchor link. _index is the default file, so you just omit that. And here are the corrected relative paths :) 
```suggestion
* [Local deployments](../../../secure/tls-encryption/)
* [YB Anywhere](../../../yugabyte-platform/security/enable-encryption-in-transit/#connect-to-a-ysql-endpoint-with-tls)
* [YB Managed](../../../yugabyte-cloud/cloud-secure-clusters/cloud-authentication/#download-your-cluster-certificate)
```
